lib/tasks/data_hygiene/publishing_delay_reporter.rb

```diff
@@ -0,0 +1,32 @@
+class Tasks::DataHygiene::PublishingDelayReporter
+  def report
+    now = Time.now
+
+    log_entries = ScheduledPublishingLogEntry.where(
+      :scheduled_publication_time.gt => now - 1.day,
+      :scheduled_publication_time.lte => now
+    )
+    delays = log_entries.map(&:delay_in_milliseconds)
+
+    unless delays.empty?
+      GovukStatsd.gauge("scheduled_publishing.aggregate.mean_ms", Stats.mean(delays))
+      GovukStatsd.gauge("scheduled_publishing.aggregate.95_percentile_ms", Stats.percentile(delays, 95))
```

Comment:
If we used GovukStatsd.timing would on all the delays would we get all these stats for free?  https://github.com/etsy/statsd/blob/master/docs/metric_types.md#timing  It might be too much though?

Reply:
Not exactly, because statsd only calculates the the stats within each reporting time interval. So you'll get data points for the mean and 90th percentile for the two or three docs that were published at the same time - it won't give you the average across an arbitrary interval like 24 hours.

But we are already recording the timing data (see #385) and it might be possible to use Graphite to calculate something useful from it. The stats won't be totally accurate because statsd has already done some aggregation, but they might be good enough. I've had a go at building a dashboard, and I'll see if Paul thinks it'll do the job: https://grafana.publishing.service.gov.uk/dashboard/db/publishing-delays-work-in-progress

Reply:
Just spoken to Paul, and he'd like us to add this cron job so we have more accurate stats.

Reply:
Yeah, I wasn't sure if `timing` would be useful, but I'd only just discovered it and wanted to use the new shiny thing everywhere ;).  If it's not fit for purpose then let's not use it.